level_5/lexer.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+#include <iostream>
+#include <stdexcept>
+#include <variant>
+#include <cctype>
+
+enum class Type {
+	INTEGER, PLUS, MINUS, MUL, DIV, END, LPARAN, RPARAN
+};
+
+class Token{
+	private:
+	Type token_type_;
+	std::variant<int, char> value_;
+	public:
+		Token(Type token_type, int value) : token_type_{token_type}, value_{value} {}
+		Token(Type token_type, char value) : token_type_{token_type}, value_{value} {}
+		Type getToken() const noexcept
```

Comment:
```suggestion
		constexpr Type getToken() const noexcept
```
The class name is already Token, and getToken is returning a Type.

Name it `getTokenType()` / `getType()` or just `type()` instead.